backend/hf_api_service.py

```
@@ -166,6 +169,33 @@ async def do_post(c):
+        else:
+            async with httpx.AsyncClient() as new_client:
+                response = await do_post(new_client)
+
+        if response.status_code == 200:
+            # Result is usually {"text": "..."}
+            data = response.json()
+            return data.get("text", "")
+        else:
+            logger.error(f"Whisper API Error: {response.status_code} - {response.text}")
+            return ""
+    except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)